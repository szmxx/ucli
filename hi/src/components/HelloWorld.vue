<!--
 * @Author: cola
 * @Date: 2023-07-15 16:03:37
 * @LastEditors: cola
 * @Description:
-->
<script setup lang="ts">
import { ref } from 'vue'
const msg = ref('hello world')
const count = ref(0)
function onClick() {
  msg.value = 'greeting'
  count.value++
}
</script>

<template>
  <h1>{{ msg }}</h1>

  <div class="card">
    <button type="button" @click="onClick">count is {{ count }}</button>
  </div>
</template>
