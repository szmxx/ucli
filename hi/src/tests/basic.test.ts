/*
 * @Author: cola
 * @Date: 2022-07-07 14:34:41
 * @LastEditors: cola
 * @Description:
 */
import { mount } from '@vue/test-utils'
import HelloWorld from '@/components/HelloWorld.vue'
describe('mount component', () => {
  it('mount component', async () => {
    const wrapper = mount(HelloWorld)
    expect(wrapper.text()).toContain('hello world')
    expect(wrapper.html()).toMatchSnapshot()
    await wrapper.get('button').trigger('click')
    expect(wrapper.text()).toContain('greeting')
  })
})
