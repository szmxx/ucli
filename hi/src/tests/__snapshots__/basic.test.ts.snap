// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`mount component > mount component 1`] = `
"<h1>hello world</h1>
<div class=\\"card\\"><button type=\\"button\\">count is 0</button></div>"
`;
